sections/discussion/discussion-chapter.tex

```diff
@@ -1,9 +1,153 @@
 \section{Discussion}\label{sec:discussion}
-Did the results work? This will all be discussed in this chapter.
+The following section will investigate the results acquired in \autoref{sec:experiments}.
+The reported values for the different metrics will be discussed in order to gain an understanding of why the different methods perform as reported, and which results are interesting.
+The comparability between different methods and datasets will also be discussed, as well as what information can be inferred in relation to the datasets.
+Finally, the usefulness of context for recommendation will be examined, as well as the performance of graph based methods.
 
+\subsection{Experimental results discussion}
+In the following subsection we discuss some of the metrics used in the experiments to evaluate the methods.
+The main goal is to answer the following question:
 \begin{itemize}
-    \item How is context used in various investigated methods?
-    \item Does context actually matter?
-    \item What did the experiments tell us?
-    \item Did the graph methods do better than the matrix based methods?
-\end{itemize}
\ No newline at end of file
+    \item Which metrics are the most interesting for comparison?
+    \item Are the experimental results comparable?
+\end{itemize}
+
+\subsubsection{The value of RMSE and MAE}
+Two of the metrics that were used for evaluating the performance of the methods that compute rating predictions were RMSE and MAE.
+One of the things to be aware of with these metrics is to be careful when using them for determining the value of the different context variables.
+For example, when a user is happy, it can be expected that every movie will be rated higher across the board which may results in a better or worse RMSE and MAE.
+The issue is that it will probably not change the top-$N$ list of recommended items, so instead one should use metrics such as Precision@N, Recall@N or F1@N to determine the value of the context variables.
```

Comment:
```suggestion
The issue is that it will probably not change the top-$N$ list of recommended items, so instead one should use metrics such as Precision@N, Recall@N, or F1@N to determine the value of the context variables.
```